src/Tuple/nMap.js

```diff
@@ -0,0 +1,68 @@
+/** @license ISC License (c) copyright 2018 original and current authors */
+/** @author Karthik Iyengar (karthikiyengar) */
+
+const curry = require('../core/curry')
+const isFunction = require('../core/isFunction')
+const isSameType = require('../core/isSameType')
+const Tuple = require('./index')
+
+const validTuple = (n, m) =>
+  isSameType(Tuple(n), m)
+
+function runMap(m, fns) {
+  const n = fns.length
+
+  if (!validTuple(n, m)) {
+    throw new TypeError(`nMap: ${n}-Tuple required`)
+  }
+
+  fns.forEach(fn => {
+    if (!isFunction(fn)) {
+      throw new TypeError('nMap: Functions required')
+    }
+  })
+
+  return m.nMap(...fns)
+}
+
+function nMap(n) {
```

Comment:
following the 🌽vention of `bimap`, `contramap`, `promap`... I am thinking this should follow suit and become `nmap`?